### ISSUE TITLE
[Required feature] Can make subspec as submodule if enable "use_modular_headers"?

### ISSUE THREAD

**Issue by alexlee002** [CocoaPods/CocoaPods#9806]
for example,   MyDemo.podspec is like this:

```ruby
# ...
s.subspec "SubLibA" do |ss|
# ...
end

s.subspec "SubLibB" do |ss|
# ...
end
```

the Podfile  import the pod:

```ruby
use_modular_headers!

# ...
pod 'MyDemo'
```

and I use the pod in my app code:

```Objective-C
@import MyDemo.SubLibA;
// ...
```

**Comment by dnkoutso** [comment]
Yes possibly a good enhancement. I think it would depend if the root spec is used or not though.

**Comment by dnkoutso** [comment]
You can also specify you own module map today using `s.module_map`.

**Comment by dnkoutso** [comment]
Actually this would be much harder now with cocoapods as subspecs can change names depending on which variants you integrate.

I dont think we will be able to support that and i recommend using your own module map.

**Comment by alexlee002** [comment]
@dnkoutso yes,  My own pods I can write my own module map to support it.

I hope  when I use a pod written by others(eg, from github),  cocoapods will automatically generate the module map with submodules according to the configs in the podspec.

**Comment by alexlee002** [comment]
> Actually this would be much harder now with cocoapods as subspecs can change names depending on which variants you integrate.
> 
> I dont think we will be able to support that and i recommend using your own module map.

@dnkoutso  Can we generate different modules for different variants?  
If we cant do that,  is that means we don't need to specify subclass in the podfile?  Instead we just add the root of pod in the podfile, and we can import sub-module in the code?

**Comment by dnkoutso** [comment]
@alexlee002 
> @dnkoutso yes, My own pods I can write my own module map to support it.

You can always add `post_install` hook to edit the pods you use to set a `MODULE_MAP` entry providing your own module map file

**Comment by alexlee002** [comment]
@dnkoutso  
Yes,  this is a solution.  But I need to manually edit the module-map file, every time when the pods reference changed in the podfile.  I hope this would be more "smart".

**Comment by dnkoutso** [comment]
I am sorry I dont fully understand. what you mean "pods reference changed in the podfile.

The `pod_target` instance in the `post_install` hook would give you pretty much everything you need  to generate a module map file on the fly and set it as part of the targets build settings.

**Comment by alexlee002** [comment]
@dnkoutso

I mean,   I don't want to manually edit the module map file every time if I change the subspecs in the Podfile.
and, maybe there would be more than one target reference the same pod, in different platform, different subspecs.  
I hope cocoapods can automatically create the different modulemap for different variants.
